src/PrestaShopBundle/Command/NamingConventionLinterCommand.php

```
@@ -0,0 +1,86 @@
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Command;
+
+use PrestaShopBundle\Routing\Linter\Exception\LinterException;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Runs naming conventions linter in the CLI
+ */
+final class NamingConventionLinterCommand extends ContainerAwareCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function configure()
+    {
+        $this
+            ->setName('prestashop:linter:naming-convention')
+            ->setDescription('Checks if Back Office routes and controllers follow naming convention.')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $adminRouteProvider = $this->getContainer()->get('prestashop.bundle.routing.linter.admin_route_provider');
+        $namingConventionLinter = $this->getContainer()
+            ->get('prestashop.bundle.routing.linter.naming_convention_linter');
+
+        $invalidRoutes = [];
+
+        /** @var Route $route */
+        foreach ($adminRouteProvider->getRoutes() as $routeName => $route) {
+            try {
+                $namingConventionLinter->lint($routeName, $route);
+            } catch (LinterException $e) {
+                $invalidRoutes[] = $routeName;
+            }
+        }
+
+        $io = new SymfonyStyle($input, $output);
+
+        if (!empty($invalidRoutes)) {
+            $io->warning(sprintf(
+                '%s routes are not following naming conventions:',
+                count($invalidRoutes)
+            ));
+            $io->listing($invalidRoutes);
+
+            return;
```

Maybe return an error code?

---

Hmmm, I don't think it's necessary. There is no error, command executes successfully either way, but can return different results. In one case, list of invalid routes is empty, in other it's not. Wdyt?

---

Here, if routes are not following conventions it should be report as errors :/
Let's try to return 1 in case of error.
after this change, if routes are not following the conventions:
```bash
$ ./bin/console myCommand
echo $?
# If value != 1 try with exception
```
Otherwise we'll not be able to add this command in our CI

---

Ok, I've updated command. But either way, we won't we able to add it to CI, because of many edge cases unless it would be informational only.

---

It was just an example :)

---

For now, it wont be added to CI. But the long-term plan is to integrate it into CI indeed